cmd/admin-handlers.go

```diff
@@ -415,6 +416,126 @@ func (a adminAPIHandlers) PerfInfoHandler(w http.ResponseWriter, r *http.Request
+	Resource   string    `json:"resource"`   // Resource contains info like bucket, object etc
+	Type       string    `json:"type"`       // Bool whether write or read lock.
+	Source     string    `json:"source"`     // Source which created the lock
+	ServerList []string  `json:"serverlist"` // RPC path of servers issuing the lock.
+	Owner      string    `json:"owner"`      // RPC path of client claiming lock.
+	ID         string    `json:"id"`         // UID to uniquely identify request of client.
+}
+
+func newLockEntry(l lockRequesterInfo, resource, server string) *LockEntry {
+	entry := &LockEntry{Timestamp: l.Timestamp, Resource: resource, ServerList: []string{server}, Owner: l.Node, Source: l.Source, ID: l.UID}
+	if l.Writer {
+		entry.Type = "Write"
+	} else {
+		entry.Type = "Read"
+	}
+	return entry
+}
+
+// LockEntries - To sort the locks
+type LockEntries []LockEntry
+
+func (l LockEntries) Len() int {
+	return len(l)
+}
+
+func (l LockEntries) Less(i, j int) bool {
+	return l[i].Timestamp.Before(l[j].Timestamp)
+}
+
+func (l LockEntries) Swap(i, j int) {
+	l[i], l[j] = l[j], l[i]
+}
+
+func topLockEntries(peerLocks []*PeerLocks) LockEntries {
+	const listCount int = 10
+	entryMap := make(map[string]*LockEntry)
+	for _, peerLock := range peerLocks {
+		if peerLock == nil {
+			continue
+		}
+		for k, v := range peerLock.Locks {
+			for _, lockReqInfo := range v {
+				if val, ok := entryMap[lockReqInfo.UID]; ok {
+					val.ServerList = append(val.ServerList, peerLock.Addr)
+				} else {
+					entryMap[lockReqInfo.UID] = newLockEntry(lockReqInfo, k, peerLock.Addr)
+				}
+			}
+		}
+	}
+	var lockEntries LockEntries
+	for _, v := range entryMap {
+		lockEntries = append(lockEntries, *v)
+	}
+	sort.Sort(lockEntries)
+	if len(lockEntries) > listCount {
+		lockEntries = lockEntries[:listCount]
+	}
+	return lockEntries
+}
+
+// PeerLocks holds server information result of one node
+type PeerLocks struct {
+	Addr  string
+	Locks GetLocksResp
+}
+
+// TopLocksHandler Get list of locks in use
+func (a adminAPIHandlers) TopLocksHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "TopLocks")
+
+	objectAPI := newObjectLayerFn()
+	if objectAPI == nil || globalNotificationSys == nil {
+		writeErrorResponseJSON(w, ErrServerNotInitialized, r.URL)
+		return
+	}
+
+	// Authenticate request
+
+	// Setting the region as empty so as the mc server info command is irrespective to the region.
+	adminAPIErr := checkAdminRequestAuthType(ctx, r, "")
+	if adminAPIErr != ErrNone {
+		writeErrorResponseJSON(w, adminAPIErr, r.URL)
+		return
+	}
+
+	thisAddr, err := xnet.ParseHost(GetLocalPeer(globalEndpoints))
+	if err != nil {
+		writeErrorResponseJSON(w, toAdminAPIErrCode(ctx, err), r.URL)
+		return
+	}
+
+	peerLocks := globalNotificationSys.GetLocks(ctx)
+	// Once we have recieved all the locks currently used information from peers
+	// add the local peer locks list as well.
+	localLocks := globalLockServer.ll.DupLockMap()
```

Comment:
globalLockServer seems to be nil in case of FS or standalone erasure mode, we need further check for this

Reply:
checking if `globalIsDistXL` is set to `true` and return `ErrMethodNotAllowed` if not.